pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -639,9 +676,30 @@ func (lb *LoadBalancer) CreateTCPLoadBalancer(name, region string, externalIP ne
 	}
 
 	status := &api.LoadBalancerStatus{}
-	status.Ingress = []api.LoadBalancerIngress{{IP: vip.Address}}
+
+	if lb.opts.FloatingNetworkId == "" {
+		status.Ingress = []api.LoadBalancerIngress{{IP: vip.Address}}
+
+		return status, nil
+	}
+
+	vipAddr, err := getVipByName(lb.network, name)
+	if err != nil {
+		return nil, err
+	}
+	floatIPOpts := floatingips.CreateOpts{
+		FloatingNetworkID: lb.opts.FloatingNetworkId,
+		PortID:            vipAddr.PortID,
```

Comment:
Aside: I suspect following https://github.com/kubernetes/kubernetes/pull/13005, that we want to use `externalIP` (now renamed to `LoadBalancerIP` elsewhere) to set this floating IP instead of the VIP (earlier in this function).  I think we should hold off on that however until we can see what the documentation leads users to expect - and what the other cloud providers end up implementing...